File: Python/Check if Circular Linked List.py
#class Node:
#    def __init__(self, data):
#        self.data = data
#        self.next = None


# your task is to complete this function
# function should return true/false or 1/0
def isCircular(head):
    # Code here
    traveller = head
    traveller = traveller.next
    while traveller:
        if traveller == head:
            return True
        traveller = traveller.next
    return False
  # Question link: https://www.geeksforgeeks.org/problems/circular-linked-list/1

File: Java/Check_if_Circular_Linked_List.java

/* Structure of LinkedList
class Node
{
	int data;
	Node next;
	Node(int d)
	{
		data = d;
		next = null;
	}
}
*/
class GfG
{
    boolean isCircular(Node head)
    {
        Node Traveller = head.next;
        
        while (Traveller != null){
            if(Traveller == head){
                return true;
            }
            Traveller = Traveller.next;
        }
        return false;
    }
} 
// Question link: https://www.geeksforgeeks.org/problems/circular-linked-list/1
